mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala

```diff
@@ -438,6 +478,27 @@ private[ml] object FeedForwardTopology {
     }
     FeedForwardTopology(layers)
   }
+
+  /**
+   * Creates a multi-layer perceptron for regression
+   *
+   * @param layerSizes sizes of layers including input and output size
```

Comment:
That's right @viirya, the output is almost always 1. There are important corner cases with multiple outputs, like [Overfeat's](http://arxiv.org/pdf/1312.6229v4.pdf) state of the art object localization predictions (see section 4.2). Instead of training 1 linear regression model on top of the network's generated features, it trains 4 linear regression models that each predict the corner of a box that surrounds an object in the image. I think that keeping the flexibility makes sense, but we'll need to add [SPARK-9120](https://issues.apache.org/jira/browse/SPARK-9120).